JYH/10Day/다트 게임.js

```diff
@@ -0,0 +1,32 @@
+function solution(dartResult) {
+    let dart = dartResult.split("");
+    let scores = [];
+    let score = 0;
+    
+    for (let i = 0; i < dart.length; i++) {
+        if (!isNaN(dart[i])) {
+            if (dart[i] === "1" && dart[i + 1] === "0") {
+                score = 10;
+                i++;
+            } else {
+                score = dart[i];
+            }
+        } else if (dart[i] === "S") {
+            scores.push(Math.pow(score, 1));
+        } else if (dart[i] === "D") {
+            scores.push(Math.pow(score, 2));
+        } else if (dart[i] === "T") {
+            scores.push(Math.pow(score, 3));
+        } else if (dart[i] === "#") {
+            scores[scores.length - 1] *= -1;
+        } else if (dart[i] === "*") {
+            scores[scores.length - 2] *= 2;
+            scores[scores.length - 1] *= 2;
+        }
+    }
+    let sum = 0;
+    for (let i = 0; i < scores.length; i++) {
+        sum += scores[i];
+    }
+    return sum;
+}
```

Comment:
```suggestion
function solution(dartResult) {
    const dart = dartResult.split("");
    const scores = [];
    let score = 0;
    let sum = 0;
    
    for (let i = 0; i < dart.length; i++) {
        if (!isNaN(dart[i])) {
            if (dart[i] === "1" && dart[i + 1] === "0") {
                score = 10;
                i++;
            } else {
                score = dart[i];
            }
        } else if (dart[i] === "S") {
            scores.push(Math.pow(score, 1));
        } else if (dart[i] === "D") {
            scores.push(Math.pow(score, 2));
        } else if (dart[i] === "T") {
            scores.push(Math.pow(score, 3));
        } else if (dart[i] === "#") {
            scores[scores.length - 1] *= -1;
        } else if (dart[i] === "*") {
            scores[scores.length - 2] *= 2;
            scores[scores.length - 1] *= 2;
        }
    }
    
    for (let i = 0; i < scores.length; i++) {
        sum += scores[i];
    }
    return sum;
}
```
참조 타입의 변수의 경우에는 웬만하면 const 상수로 선언하는게 좋습니다!
또한 sum같은 변수의 경우에는 코드의 최 상단에 위치해주는게 가독성이 올라갑니다👍